diff --git a/dist/theme/components/Search/SearchPanel.js b/dist/theme/components/Search/SearchPanel.js
index f57c8ff75e973991d95dfd34c5784dd12b776c02..944c5198c91203bd5edc3b3eccfb614e46a4ad19 100644
--- a/dist/theme/components/Search/SearchPanel.js
+++ b/dist/theme/components/Search/SearchPanel.js
@@ -68,7 +68,7 @@ function SearchPanel({ focused, setFocused }) {
                     break;
                 case KEY_CODE.ARROW_DOWN:
                     if (e.isComposing) return;
-                    if (focused) {
+                    if (focused && e.target === searchInputRef.current && currentSuggestions.length > 0) {
                         e.preventDefault();
                         if (currentSuggestions && currentRenderType === RenderType.Default) {
                             setCanScroll(true);
@@ -78,7 +78,7 @@ function SearchPanel({ focused, setFocused }) {
                     break;
                 case KEY_CODE.ARROW_UP:
                     if (e.isComposing) return;
-                    if (focused) {
+                    if (focused && e.target === searchInputRef.current && currentSuggestions.length > 0) {
                         e.preventDefault();
                         if (currentRenderType === RenderType.Default) {
                             const currentSuggestionsLength = currentSuggestions.length;
@@ -89,9 +89,11 @@ function SearchPanel({ focused, setFocused }) {
                     break;
                 case KEY_CODE.ENTER:
                     if (e.isComposing) return;
-                    if (currentSuggestionIndex >= 0 && currentRenderType === RenderType.Default) {
+                    if (focused && e.target === searchInputRef.current && currentSuggestionIndex >= 0 && currentRenderType === RenderType.Default) {
                         const flatSuggestions = Array.from(normalizeSuggestions(currentSuggestions).values()).flat();
                         const suggestion = flatSuggestions[currentSuggestionIndex];
+                        if (!suggestion) return;
+                        e.preventDefault();
                         navigate(suggestion.link);
                         clearSearchState();
                     }
